### PR TITLE
gh-95913: Edit & expand Optimizations in 3.11 WhatsNew

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
Part of #95913

I've added Sphinx markup and cross references to the Optimizations section of the Python What's New document, copyedited the text to fix grammar errors, read better and be slightly clearer, and sorted it by language, then builtins, then standard library (sorted by module name).

Additionally, I've added a handful of additional optimizations to common functions with documented substantial (>=15%) performance increases mentioned in the changelog (that were unrelated to Faster CPython), and consistent with the existing optimizations mentioned here and in previous verisons. These are:

* #90564 Integer division ``//`` 20% faster 
* #68264 `sum()` 30% faster
* #91165 `list.append` 15% faster / list comprehensions 20-30% faster
* #91487 `asyncio.DatagramProtocol` >100x faster on large files
* #90415 `statistics` functions `mean`, `variance` and `stdev` 2x faster and less memory

* Issue: gh-95913

### CODE REVIEW

File: Doc/whatsnew/3.11.rst
Comment (JelleZijlstra) [inline Doc/whatsnew/3.11.rst]: Wasn't this one part of Faster CPython?

**Review by JelleZijlstra — APPROVED** [review]

File: Doc/whatsnew/3.11.rst
Comment (Fidget-Spinner) [inline Doc/whatsnew/3.11.rst]: Could you move this to faster cpython and attribute Brandt please? It's `gh-91404`.

File: Doc/whatsnew/3.11.rst
Comment (CAM-Gerlach) [reply Doc/whatsnew/3.11.rst]: I wasn't sure about that either, especially since it was @markshannon who implemented it, but it at least wasn't listed there (whereas another exception-related change, for catching an exception when it is raised, is). I'm happy to move it if so.

@markshannon @Fidget-Spinner could you confirm? If it was, I can just remove it here and add it to the Misc section in #98429 , next to the other exception change and alongside the `re` change mentioned below. Thanks!

File: Doc/whatsnew/3.11.rst
Comment (CAM-Gerlach) [reply Doc/whatsnew/3.11.rst]: Sure, I deleted it here and added it with attribution to #98429 . I actually added the attribution originally but hesitated, just in case for some reason it Brandt omitted it intentionally.

File: Doc/whatsnew/3.11.rst
Comment (markshannon) [reply Doc/whatsnew/3.11.rst]: Yes, it was part of the Faster CPython project.

Comment (CAM-Gerlach) [reply Doc/whatsnew/3.11.rst]: Thanks, I went ahead and moved it too

File: Doc/whatsnew/3.11.rst
Comment (markshannon) [inline Doc/whatsnew/3.11.rst:1145]: `sum()` is still much slower than 2.7.
Maybe we shouldn't mention this, until it is fixed in 3.12

Comment (CAM-Gerlach) [reply Doc/whatsnew/3.11.rst:1145]: Sure, though on the other hand, it still a fairly substantial improvement over 3.10, the previous feature version, which seems to me is the main focus of this section—at least in the past, I'm not sure optimizations were only mentioned if they exceeded 2.7's speed for something, since that likely wasn't the case for quite a few of them. That said, I'm certainly not the expert here (unlike you all), and will of course defer to the judgement of @pablogsal as RM and you and rest of the core dev performance folks on whether this should be included, if you think its best left out.

Comment (terryjreedy) [reply Doc/whatsnew/3.11.rst:1145]: On discuss - release, Pable said "I don’t think there is anything wrong mentioning that is faster than the version in 3.10. After all there was work here to make the situation better and if true, I don’t see why a considerable improvement over the previous version should not be mentioned".  Sum is not new, but since we have an optimization section, I agree.

**Pull request merged**
